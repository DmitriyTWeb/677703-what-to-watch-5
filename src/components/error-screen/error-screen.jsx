import React from "react";
import PropTypes from "prop-types";
import {connect} from "react-redux";
import {Link} from "react-router-dom";
import {AppRoute} from "../../const.js";
import {getError} from "../../store/selectors/selectors.js";
import {resetError} from "../../store/action.js";

const erorrStyle = {
  display: `flex`,
  flexDirection: `column`,
  justifyContent: `center`,
  alignItems: `center`,
  position: `fixed`,
  top: `0`,
  left: `0`,
  right: `0`,
  height: `auto`,
  zIndex: `10`,
  backgroundColor: `red`,
  paddingBottom: `30px`,
};

const styleP = {
  textAlign: `center`,
};


const ErrorScreen = (props) => {
  const {
    error,
    resetErrorAction,
  } = props;

  if (!error) {
    return null;
  }
  return <React.Fragment>
    <section style={erorrStyle}>
      <h1>Error</h1>
      <h2>Well poo. Sadly, the website had a bad day.</h2>

      <p style={styleP}>{error.message}</p>
      <p style={styleP}>{error.stack}</p>

      <button onClick={resetErrorAction}>
        RESET ERROR
      </button>
    </section>
  </React.Fragment>;
};

ErrorScreen.propTypes = {
  error: PropTypes.object,
  resetErrorAction: PropTypes.func.isRequired,
};

const mapStateToProps = (state) => ({
  error: getError(state),
});
const mapDispatchToProps = (dispatch) => ({
  resetErrorAction() {
    dispatch(resetError());
  }
});

export {ErrorScreen};
export default connect(mapStateToProps, mapDispatchToProps)(ErrorScreen);
