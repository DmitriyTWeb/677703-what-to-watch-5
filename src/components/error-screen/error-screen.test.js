import React from "react";
import renderer from "react-test-renderer";
import {ErrorScreen} from "./error-screen";

const noop = () => { };

it(`Should ErrorScreen render correctly`, () => {
  const tree = renderer.create(
      <ErrorScreen
        error={{}}
        resetErrorAction={noop}
      />
  ).toJSON();

  expect(tree).toMatchSnapshot();
});

