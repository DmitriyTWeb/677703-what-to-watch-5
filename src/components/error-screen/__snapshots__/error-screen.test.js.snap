// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should ErrorScreen render correctly 1`] = `
<section
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "red",
      "display": "flex",
      "flexDirection": "column",
      "height": "auto",
      "justifyContent": "center",
      "left": "0",
      "paddingBottom": "30px",
      "position": "fixed",
      "right": "0",
      "top": "0",
      "zIndex": "10",
    }
  }
>
  <h1>
    Error
  </h1>
  <h2>
    Well poo. Sadly, the website had a bad day.
  </h2>
  <p
    style={
      Object {
        "textAlign": "center",
      }
    }
  />
  <p
    style={
      Object {
        "textAlign": "center",
      }
    }
  />
  <button
    onClick={[Function]}
  >
    RESET ERROR
  </button>
</section>
`;
